process/smartContract/hooks/blockChainHook_test.go

```diff
@@ -583,10 +1480,40 @@ func TestBlockChainHookImpl_GetESDTToken(t *testing.T) {
 		require.NotNil(t, esdtData)
 		assert.Equal(t, emptyESDTData, esdtData)
 	})
+	t.Run("error unmarshall", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockBlockChainHookArgs()
+		invalidUnmarshalledData := []byte("invalid data")
+		args.Accounts = &stateMock.AccountsStub{
+			GetExistingAccountCalled: func(addressContainer []byte) (vmcommon.AccountHandler, error) {
+				require.Equal(t, address, addressContainer)
+				account := mock.NewAccountWrapMock(address)
+				_ = account.DataTrieTracker().SaveKeyValue(completeEsdtTokenKey, invalidUnmarshalledData)
+
+				return account, nil
+			},
+		}
+		errMarshaller := errors.New("error marshaller")
+		args.Marshalizer = &testscommon.MarshalizerStub{
+			UnmarshalCalled: func(obj interface{}, buff []byte) error {
+				require.Equal(t, emptyESDTData, obj)
+				require.Equal(t, invalidUnmarshalledData, buff)
+				return errMarshaller
+			},
+		}
+
+		bh, _ := hooks.NewBlockChainHookImpl(args)
+		bh.SetFlagOptimizeNFTStore(false)
+
+		esdtData, err := bh.GetESDTToken(address, token, nonce)
+		require.Nil(t, esdtData)
```

Comment:
not related to this PR, but we have some typos in this file. can you replace `addres` with `address` ?

Reply:
Right, it should've been `addressContainer` actually. Refactoed it